packages/core/src/odata/common/entity-deserializer.ts

```diff
@@ -192,6 +196,21 @@ export function entityDeserializer(
       );
   }
 
+  function deserializeComplexTypeProperty(
```

Comment:
ah, readability +1

Reply:
Question: Applies also to the serialize analogon below. Should we for the last step return the edmToTs or should we also do a check if it is edm and throw an exception otherwise so that we have a better error handling?

Reply:
Hmm, what would a check like this look like? Pseudocode: `propertyMetadata.type is edmType`? For this our generation must be broken, I guess.
`edmToTs` is quite forgiving here. If there is no serializer for this type, we just pass the value as is, if there is and the format is not correct, the corresponding converters fail. I don't have any strong opinion here ¯\_(ツ)_/¯

Reply:
ok if the edm is forgiving we can use it as a fallback.